Makefile

```diff
@@ -1,32 +1,43 @@
-CODE_ROOT := er_aws_msk
+# Keep in sync with the Python version in the Dockerfile
+PYTHON_VERSION ?= 3.11
+SITE_PACKAGES_DIR ?= $(shell python3 -c 'import sysconfig; print(sysconfig.get_paths()["purelib"])')
 CONTAINER_ENGINE ?= $(shell which podman >/dev/null 2>&1 && echo podman || echo docker)
-IMAGE_NAME := quay.io/app-sre/er-aws-msk
-IMAGE_TAG := $(shell git describe --tags)
-ifeq ($(IMAGE_TAG),)
-	IMAGE_TAG = pre
-endif
-BUILD_ARGS := CODE_ROOT=$(CODE_ROOT) POETRY_VERSION=1.8.3 IMAGE_NAME=$(IMAGE_NAME) IMAGE_TAG=$(IMAGE_TAG)
 
 .PHONY: format
 format:
-	poetry run ruff check
-	poetry run ruff format
+	uv run ruff check
+	uv run ruff format
+
+.PHONY: image_tests
+image_tests:
+	# test /tmp must be empty
+	[ -z "$(shell ls -A /tmp)" ]
+	# validate_plan.py must exist
+	[ -f "validate_plan.py" ]
+
+.PHONY: code_tests
+code_tests:
+	uv run ruff check --no-fix
+	uv run ruff format --check
+	uv run mypy
+	uv run pytest -vv --cov=er_aws_msk --cov-report=term-missing --cov-report xml
+
+.PHONY: dependency_tests
+dependency_tests:
+	python -c "import cdktf_cdktf_provider_random"
+	python -c "import cdktf_cdktf_provider_aws"
 
 .PHONY: test
-test: build
-	$(CONTAINER_ENGINE) build -t $(IMAGE_NAME)-test $(foreach arg,$(BUILD_ARGS),--build-arg $(arg)) -f dockerfiles/Dockerfile.test .
-	$(CONTAINER_ENGINE) run --rm --entrypoint poetry $(IMAGE_NAME)-test run ruff check --no-fix
-	$(CONTAINER_ENGINE) run --rm --entrypoint poetry $(IMAGE_NAME)-test run ruff format --check
-	$(CONTAINER_ENGINE) run --rm --entrypoint poetry $(IMAGE_NAME)-test run mypy
-	$(CONTAINER_ENGINE) run --rm --entrypoint poetry $(IMAGE_NAME)-test run pytest -vv --cov=$(CODE_ROOT) --cov-report=term-missing --cov-report xml
+test: image_tests code_tests dependency_tests
 
 .PHONY: build
 build:
-	$(CONTAINER_ENGINE) build -t $(IMAGE_NAME):${IMAGE_TAG} $(foreach arg,$(BUILD_ARGS),--build-arg $(arg)) -f dockerfiles/Dockerfile .
+	$(CONTAINER_ENGINE) build --progress plain -t er-aws-msk:test .
 
-.PHONY: push
-push:
-	$(CONTAINER_ENGINE) push ${IMAGE_NAME}:${IMAGE_TAG}
 
-.PHONY: deploy
-deploy: build test push
+.PHONY: dev
+dev:
+	uv sync --python $(PYTHON_VERSION)
+	# Prepare local development environment
+	$(CONTAINER_ENGINE) run --rm -it -v $(PWD)/:/home/app/src -v $(PWD)/.gen:/cdktf-providers:z quay.io/redhat-services-prod/app-sre-tenant/er-base-cdktf-main/er-base-cdktf-main:cdktf-0.20.9-tf-1.6.6-py-3.11-v0.3.0 cdktf-provider-sync /cdktf-providers
```

Comment:
Do we need to use this tag?

Reply:
I changed it to `latest`, which is sufficient for the development environment.